spec/ysh-stdlib.test.sh

```diff
@@ -97,3 +97,93 @@ echo status=$_status
 42
 status=0
 ## END
+
+#### any
+source --builtin list.ysh
+
+json write (any([]))
+json write (any([true]))
+json write (any([false]))
+json write (any([true, false]))
+json write (any([false, true]))
+json write (any([false, false]))
+json write (any([false, true, false]))
+json write (any([false, false, null, false]))  # null is falsey
+## STDOUT:
+false
+true
+false
+true
+true
+false
+true
+false
+## END
+
+#### all
+source --builtin list.ysh
+
+json write (any([]))
+json write (any([true]))
+json write (any([false]))
+json write (any([true, true]))
+json write (any([true, false]))
+json write (any([false, true]))
```

Comment:
could have some test cases with "foo" and "" and so forth

It will follow the `Bool()` builtin, which we don't have yet I think (or at least not in C++)